tools/rpkg/src/duckdbr.cpp

```diff
@@ -1169,6 +1170,113 @@ SEXP duckdb_unregister_R(SEXP connsexp, SEXP namesexp) {
 	return R_NilValue;
 }
 
+class RArrowTabularStreamFactory {
+public:
+	RArrowTabularStreamFactory(SEXP export_fun_p, SEXP arrow_tabular_p)
+	    : arrow_tabular(arrow_tabular_p), export_fun(export_fun_p) {};
+	static unique_ptr<ArrowArrayStreamWrapper> Produce(uintptr_t factory_p) {
+		RProtector r;
+		int err;
+		auto factory = (RArrowTabularStreamFactory *)factory_p;
+
+		// export the arrow tabular to a stream pointer
+		auto export_call = r.Protect(Rf_lang2(factory->export_fun, factory->arrow_tabular));
+		SEXP stream_ptr_sexp = r.Protect(R_tryEval(export_call, R_GlobalEnv, &err));
+		if (err) {
+			Rf_error("Failed to produce arrow stream");
+		}
+		// TODO this stream pointer leaks?
```

Comment:
Yes, you want to call `delete_arrow_array_stream()` after you've ingested the data in duckdb. Here's how we do it when we transfer a RecordBatchReader from R to Python: https://github.com/apache/arrow/blob/master/r/R/python.R#L169-L175

(This is one reason I was asking about evaluating `export_fun` in R rather than C, so that you can set the `on.exit(delete_arrow_array_stream(stream_ptr))`)

Reply:
I think since we already have an arrow stream internally, I am now trying to pass that directly to `ExportRecordBatchReader`